### ISSUE TITLE
Colormapping multiple images

### ISSUE THREAD

**Issue by ianthomas23** [bokeh/bokeh#12355]
### Discussed in https://github.com/bokeh/bokeh/discussions/12354

<div type='discussions-op-text'>

<sup>Originally posted by **ianthomas23** September  7, 2022</sup>
`Figure.image()` can take multiple images (multiple 2D arrays) that are colormapped using the same `ColorMapper`. This can give rise to some strange results. Consider this example which has one array with values in the range 0 to 1, and a second with values in the range 1 to 2.
```python
from bokeh.models import ColorBar, LinearColorMapper
from bokeh.plotting import figure, show
import numpy as np

rng = np.random.default_rng(3841)
n = 20
data0 = rng.random((n, n))
data1 = data0 + 1.0

p = figure(width=600, height=300)
mapper = LinearColorMapper(palette="Spectral10")

p.image(image=[data0, data1], x=[0, 1.1], y=0, dw=1, dh=1, color_mapper=mapper)

cbar = ColorBar(color_mapper=mapper)
p.add_layout(cbar, place="right")

show(p)
```
This produces
![Screenshot 2022-09-07 at 12-29-34 Bokeh Plot](https://user-images.githubusercontent.com/580326/188868116-86d84864-f132-4643-8953-cc14c9ac9b13.png)
Here each array is separately colormapped so they both use the full palette and are rendered with the same colors. The colorbar shows the mapping used for the second image so the limits are approximately 1 to 2. If you swap the order of the 2 arrays then the output is the same except that the colorbar limits are approximately 0 to 1.

I can get what I consider to be the correct output using e.g.
```python
mapper = LinearColorMapper(palette="Spectral10", low=0, high=2)
```
which produces
![Screenshot 2022-09-07 at 12-29-54 Bokeh Plot](https://user-images.githubusercontent.com/580326/188868984-b490da6a-827c-458f-b6b0-b0438a6bd6e5.png)
which essentially forces the colormapper to ignore the actual data values. This wouldn't work for `EqHistColorMapper` as it always needs to scan the data.

This isn't really a problem caused by `Figure.image` taking multiple arrays, you can obtain the same result using two `Figure.image` calls that take different arrays but the same colormapper. But then at least you can specify different colormappers for each `Figure.image` call whereas the single `Figure.image` call with multiple arrays is obliged to use a single colormapper.

`ContinuousColorMapper` has a `domain` property that I think is relevant here, but if I replace the single `p.image` call above with
```python
r0 = p.image(image=[data0], x=0.0, y=0, dw=1, dh=1, color_mapper=mapper)
r1 = p.image(image=[data1], x=1.1, y=0, dw=1, dh=1, color_mapper=mapper)
mapper.domain = [r0, r1]
```
then I still see the same result.

I am not sure if this is a bug or a documentation issue, or user error, or whatever, hence I have created it as a discussion.</div>

**Comment by ianthomas23** [comment]
See discussion #12354 for confirmation that this is a bug.

**Comment by bryevdv** [comment]
I think https://github.com/bokeh/bokeh/issues/4847 was the original mention of this. GIven this issue has more discussion and  information, I would propose closing the old issue as a duplicate.

**Comment by ianthomas23** [comment]
I agree with closing the old issue. I like the fact that it was put on a `short-term` milestone back in 2016!

**Comment by bryevdv** [comment]
Intentions very often outstrip resources :) FWIW whenever I have thought about this, I have been surprised more people never complained about it. 🤷

**Comment by ianthomas23** [comment]
I live with a geologist. Any time interval less than 1000 years is "short term"!
